apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -63,3 +80,33 @@ export const WarningLabel = ({children}) => (
 WarningLabel.propTypes = {
   children: React.PropTypes.node
 };
+
+/**
+ * Given a search query, generate a results list of animationProps objects that
+ * can be displayed and used to add an animation to the project.
+ * @param {string} searchQuery - text entered by the user to find an animation
+ */
+function searchAnimations(searchQuery) {
+  // Make sure to generate the search regex in advance, only once.
+  // Search is case-insensitive
+  // Match any word boundary or underscore followed by the search query.
+  // Example: searchQuery "bar"
+  //   Will match: "barbell", "foo-bar", "foo_bar" or "foo bar"
+  //   Will not match: "foobar", "ubar"
+  const searchRegExp = new RegExp('(?:\\b|_)' + searchQuery, 'i');
+
+  // Generate the set of all results associated with all matched aliases
+  const resultSet = Object.keys(animationLibrary.aliases)
+      .filter(alias => searchRegExp.test(alias))
+      .reduce((resultSet, nextAlias) => {
+        return resultSet.union(animationLibrary.aliases[nextAlias]);
+      }, Immutable.Set());
```

Comment:
I'm trying out Immutable.JS `Set` here, so I don't have to think about removing duplicates.  Not really sure if there's a performance benefit.